tests/Fixer/Basic/NumericLiteralSeparatorFixerTest.php

```diff
@@ -0,0 +1,138 @@
+    }
+
+    public static function provideFixCases(): iterable
+    {
+        $cases = [
+            'decimal' => [
+                '1234' => '1_234',
+                '-1234' => '-1_234',
+                '12345' => '12_345',
+                '123456' => '123_456',
+            ],
+            'binary' => [
+                '0b0101010001101000' => '0b01010100_01101000',
+                '0b01010100011010000110010101101111' => '0b01010100_01101000_01100101_01101111',
+                '0b110001000' => '0b1_10001000',
+            ],
+            'float' => [
+                '1234.5' => '1_234.5',
+                '1.2345' => '1.234_5',
+                '1234e5' => '1_234e5',
+                '1234E5' => '1_234E5',
+                '1e2345' => '1e2_345',
+                '1234.5678e1234' => '1_234.567_8e1_234',
+                '1.1e-1234' => '1.1e-1_234',
+                '1.1e-12345' => '1.1e-12_345',
+                '1.1e-123456' => '1.1e-123_456',
+            ],
+            'hexadecimal' => [
+                '0x42726F776E' => '0x42_72_6F_77_6E',
+                '0X42726F776E' => '0X42_72_6F_77_6E',
+                '0x2726F776E' => '0x2_72_6F_77_6E',
+                '0x1234567890abcdef' => '0x12_34_56_78_90_ab_cd_ef',
+                '0X1234567890ABCDEF' => '0X12_34_56_78_90_AB_CD_EF',
+                '0x1234e5' => '0x12_34_e5',
+            ],
+            'octal' => [
+                '012345' => '012_345',
+                '0123456' => '0123_456',
+                '01234567' => '01_234_567',
+            ],
+        ];
+
+        if (\PHP_VERSION_ID >= 8_01_00) {
+            // Test new 8.1 Octal notation
+            $cases['octal'] += [
+                '0o12345' => '0o12_345',
+                '0o123456' => '0o123_456',
+            ];
+        }
+
+        foreach ($cases as $pairsType => $pairs) {
+            foreach ($pairs as $withoutSeparator => $withSeparator) {
+                yield "{$pairsType}#{$withoutSeparator}" => [
+                    sprintf('<?php echo %s;', $withSeparator),
+                    sprintf('<?php echo %s;', $withoutSeparator),
+                ];
+            }
+        }
+    }
+
+    /**
+     * @dataProvider provideFixNoOverrideExistingCases
+     */
+    public function testFixNoOverrideExisting(string $expected, ?string $input = null): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->fixer->configure(['override_existing' => false]);
+        $this->doTest($expected, $input);
+    }
+
+    public static function provideFixNoOverrideExistingCases(): iterable
+    {
+        yield 'no_override_existing#01' => [
+            sprintf('<?php echo %s;', '0B01010100_01101000'),
+            sprintf('<?php echo %s;', '0B01010100_01101000'),
+        ];
+
+        yield 'no_override_existing#02' => [
+            sprintf('<?php echo %s;', '70_10_00'),
+            sprintf('<?php echo %s;', '70_10_00'),
+        ];
```

Comment:
Cases for scenarios where fixer shouldn't be applied don't need second argument (first one one is treated as an input *and* the output) + proper `--testdox` names

```suggestion
        yield 'no override binary' => [
            sprintf('<?php echo %s;', '0B01010100_01101000'),
        ];

        yield 'no override PHP-version-like decimals' => [
            sprintf('<?php echo %s;', '70_10_00'),
        ];
```